lib/opencontrol/parse.go

```diff
@@ -19,26 +20,30 @@ const (
 	ErrMalformedBaseYamlPrefix = "Unable to parse yaml data"
 )
 
+// YAMLParser is the concrete implementation of parsing different schema versions in YAML format.
+type YAMLParser struct{}
+
 // Parse will try to parse the data and determine which specific version of schema to further parse.
-func Parse(parser base.SchemaParser, data []byte) (base.OpenControl, error) {
+func (parser YAMLParser) Parse(data []byte) (common.OpenControl, error) {
```

Comment:
not here because there are no member variables of YAMLParser that are being modified that we want to preserve. Let's look at the declaration of the struct
```go
type YAMLParser struct{}
```

Now, if we want to, for example, keep track of the number times we parse with the parser, our declaration would look like 

```go
type YAMLParser struct{
    Count int
}
```

Now, if we didn't have the pointer (as-is now)
```go
func (parser YAMLParser) Parse {
    // stuff
    parser.Count++
}
```
Count would remain 0 forever.

and if you ran with the pointer, the count would update.
Example: https://play.golang.org/p/F-OVb1uVE2


tl;dr There's no state that needs to be carried between Parsing